app/deploy_test.go

```diff
@@ -869,3 +869,25 @@ func (s *S) TestRebuild(c *check.C) {
 	c.Assert(writer.String(), check.Equals, "Rebuild deploy called")
 	c.Assert(imgID, check.Equals, "app-image")
 }
+
+func (s *S) TestRollbackUpdate(c *check.C) {
```

Comment:
You should test more scenarios (or add more test funcs) like enabling the image after it's been disabled.